src/Agent.Worker/TaskRunner.cs

```diff
@@ -114,6 +115,21 @@ public async Task RunAsync()
                 // Setup container stephost and the right runtime variables for running job inside container.
                 if (stepTarget is ContainerInfo containerTarget)
                 {
+                    if (Stage == JobRunStage.PostJob
+                        && AgentKnobs.SkipPostExeceutionIfTargetContainerStopped.GetValue(ExecutionContext).AsBoolean())
+                    {
+                        // Check that the target container is still running, if not Skip task execution
+                        IDockerCommandManager dockerManager = HostContext.GetService<IDockerCommandManager>();
+                        bool isContainerRunning = await dockerManager.IsContainerRunning(ExecutionContext, containerTarget.ContainerId);
+                        
+                        if (!isContainerRunning)
+                        {
+                            ExecutionContext.Result = TaskResult.Skipped;
+                            ExecutionContext.ResultCode = "Target container has been stopped, task execution will be skipped";
```

Comment:
Just for visibility - could you please add container name to this message?

Reply:
Added